### PR TITLE
Add example to the tutorial for extending a widget's renderer

### PR DESCRIPTION
**Pull request by justdan96**
As title, I wasn't exactly sure how to phrase things so if anything needs editing please let me know!

### CODE REVIEW

File: tutorial/extending-widgets.md [hunk 1]
```diff
@@ -68,3 +68,91 @@ func main() {
...
+```
+
+We can test out our new extended widget using a simple program such as the below.
+
+```go
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/app"
+	"fyne.io/fyne/widget"
+	"image/color"
```
Comment (andydotxyz) [inline tutorial/extending-widgets.md:146]: standard packages should go before fyne

File: tutorial/extending-widgets.md [hunk 2]
```diff
@@ -68,3 +68,91 @@ func main() {
...
+	skinnedentry.ExtendBaseWidget(skinnedentry)
+	entryrenderer := skinnedentry.Entry.CreateRenderer()
+	return &skinnedEntryRenderer{
+		skinnedentry: skinnedentry,
+		entryrenderer: entryrenderer,
+	}
+}
+
+type skinnedEntryRenderer struct {
+	skinnedentry *skinnedEntry
```
Comment (andydotxyz) [inline tutorial/extending-widgets.md:107]: I think recommended in Go would be `skinnedEntry` so the word break is clearer

File: tutorial/extending-widgets.md [hunk 3]
```diff
@@ -68,3 +68,91 @@ func main() {
 	w.ShowAndRun()
 }
 ```
+
+We can also extend a widget and its renderer, allowing you to override its
```
Comment (andydotxyz) [inline tutorial/extending-widgets.md:72]: The "and its" here does not seem needed - really we are only covering extending a renderer in this section

File: tutorial/extending-widgets.md [hunk 4]
```diff
@@ -68,3 +68,91 @@ func main() {
 	w.ShowAndRun()
 }
 ```
+
```
Comment (andydotxyz) [inline tutorial/extending-widgets.md:71]: To match the heading at the top, how about adding:

```
## Extending a Widget Renderer
```

or similar?
At some point we may rearrange the tutorials - this makes it clear it's a topic change

File: tutorial/extending-widgets.md [hunk 5]
```diff
@@ -68,3 +68,91 @@ func main() {
...
+We can also extend a widget and its renderer, allowing you to override its 
+default appearance. For instance if we want to override the default background 
+colour of the Entry widget, we first have to extend the Entry base widget
```
Comment (andydotxyz) [inline tutorial/extending-widgets.md:74]: I'd remove "base" here - it could be confused with `BaseWidget`

**Review by andydotxyz — CHANGES_REQUESTED** [review]
This is a great addition, thanks.
A few comments about the wording and formatting but after a quick revision should be good to include.

**Comment by justdan96** [comment]
I've made the amendments suggested, let me know if you think it looks ready now.

**Review by andydotxyz — APPROVED** [review]
Great thanks

**Pull request merged**